src/shared/utils/backoffFetch.ts

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Defines all pending methods.
+const pendingMethods: (() => void)[] = []
+
+// Defines the maximum backoff.
+const maxBackoff = 512
+
+// Cancels all pending requests.
+export const cancel = () => {
+    for (const m of pendingMethods) m()
+    pendingMethods.length = 0
+}
+
+// A fetch client that will behave exactly like fetch except it will backoff for 429/5XX errors.
+export default (input: RequestInfo, init?: RequestInit): Promise<Response> => new Promise(async (res, rej) => {
+    // Defines if it is active.
+    let active = true
+
+    // Defines the method to kill the promise.
+    pendingMethods.push(() => active = false)
```

Comment:
We should also `console.info` that the request has been terminated, similar to how we're logging backoffs.